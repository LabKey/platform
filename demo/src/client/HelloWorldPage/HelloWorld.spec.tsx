

import * as React from "react";
import {mount} from "enzyme";
import toJson from "enzyme-to-json";
import {App} from "./HelloWorld";

describe('HelloWorld', () => {

    test('HelloWorld Text', () => {

        const helloWorld  = mount(<App />);

        // Verify Hello World! text
        const helloWorldSpan = helloWorld.find({className: 'world-highlight'});
        expect(helloWorldSpan.length).toEqual(1);
        expect(helloWorldSpan.text()).toEqual("Hello World!");

        // Verify snapshot
        expect(toJson(helloWorld)).toMatchSnapshot();
        helloWorld.unmount();
    });
});