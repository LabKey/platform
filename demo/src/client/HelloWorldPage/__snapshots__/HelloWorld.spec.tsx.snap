// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld HelloWorld Text 1`] = `
<App>
  <p>
    <span
      className="world-highlight"
    >
      Hello World!
    </span>
     This is a simple test page to show a LabKey HTML view built with React.
  </p>
</App>
`;
